### PR TITLE
[ci] Changelog: split workflows for PR updates and milestone assignments

### PR DESCRIPTION
**Pull request by shvgn**
## Description

Split github workflows for the cases when PR changes and when milestone is reassigned. 

## Why do we need it, and what problem does it solve?

Reduces the amount of requests to Github API and saves us some rate limits

## Changelog entries

```changes
section: ci
type: fix 
summary: Splitted workflows for PR edits and Milestone assignments
impact: Reduced the number of changelog re-generation
```

### CODE REVIEW

File: .github/workflows/changelog-by-milestone.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          count="$(echo $milestones | jq '. | length')"
+
+          echo "::set-output name=list::${milestones}"
+          echo "::set-output name=count::${count}"
+
+    outputs:
+      list: ${{ steps.milestones.outputs.list }}
+      count: ${{ steps.milestones.outputs.count }}
+
+  chanegelogs:
```
Comment (diafour) [inline .github/workflows/changelog-by-milestone.yml:90]:
```suggestion
  changelogs:
```

File: .github/workflows/changelog-by-pull.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        id: pr
+        uses: actions/github-script@v5
+        with:
+          result-encoding: string
+          script: |
+            return "ok"
+    outputs:
+      ok: ${{ steps.pr.outputs.result }}
+
+  chanegelog:
```
Comment (diafour) [inline .github/workflows/changelog-by-pull.yml:37]:
```suggestion
  changelog:
```

File: .github/workflows/changelog-by-pull.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    #   - and its milestone is open
+    if: |
+      (
+        github.event.pull_request &&
+        !contains(github.event.pull_request.labels.*.name, 'changelog')
+      ) && (
+        github.event.pull_request.state == 'closed'         &&
+        github.event.pull_request.merged                    &&
+        github.event.pull_request.milestone.state == 'open'
+      )
```
Comment (diafour) [inline .github/workflows/changelog-by-pull.yml:25]: I don't get it, why this `if` needs a separate job and not in a `changelog` job?

**Review by diafour — CHANGES_REQUESTED** [review]

File: .github/workflows/changelog-by-pull.yml (hunk 3 above)
Comment (shvgn) [reply .github/workflows/changelog-by-pull.yml:25]: Agreed, separate condition job is redundant. Fixed.

**Review by diafour — APPROVED** [review]

**Pull request merged**